e2e/ctl_v2_test.go

```diff
@@ -312,6 +312,30 @@ func TestCtlV2AuthWithCommonName(t *testing.T) {
 	}
 }
 
+func TestCtlV2AuthWithGuestRole(t *testing.T) {
```

Comment:
I do not think this is the correct test.

We should set up a cluster with tls + client Auth checking. Then a client with cert that has no valid CN name should be recognized as guest.